### PR TITLE
Markdown conversion

### PR DESCRIPTION
**Pull request by Pyrrh**
Created java script to convert markdown. Potentially not fully completed but need to see how markdown for post 982 converts and decide from there.

To be merged AFTER the tutorial branch

### CODE REVIEW

**Comment by sreynen** [comment]
How do I run the Java code? From reading it, I gather that it's converting files like `post_982.txt` to files like `Reality as a Process.md` but I'm not familiar enough with Java to know how to see that in action and my attempts to follow instructions I've found have all resulted in error messages.

Pulling the relevant data out of a single row of a MySQL dump like this seems like a pretty error-prone strategy, but that's probably fine as long as we have some way to detect errors when they happen. E.g. presumably Gatsby will throw an error if it can't parse a Markdown file, and then we can just manually clean up whatever's wrong with that file.

But just in case it ends up being relevant, I think this would be a little cleaner to export the database table to a CSV and then parse that CSV instead, relying on a more standard format with standard parsers.

**Comment by Pyrrh** [comment]
To compile & run the java: first compile "javac Converter.java" then run it "java Converter <filename>"

right now my plan was just to test the one post and make sure it works, and then refactor the code to take the entire set of comma-delimited entries and parse all of them at once. I'm not sure that a CSV is necessary. If you still disagree then I'm happy to approach it from that direction instead.

File: markdown_posts/markdown_converter/Converter.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+//update mistranslated characters
+          split[i] = replaceSubstring (split[i], "&#8217;", "'");
+          split[i] = replaceSubstring (split[i], "â€“", "-");
+          split[i] = replaceSubstring (split[i], "â€™", "'");
+          split[i] = replaceSubstring (split[i], "\\\"", "\"");
+          split[i] = replaceSubstring (split[i], "&#8220;", "\"");
+          split[i] = replaceSubstring (split[i], "&#8221;", "\"");
+          split[i] = replaceSubstring (split[i], "\\\'", "'");
+          //this one will convert browser newlines to markdown newlines
+          split[i] = replaceSubstring (split[i], "\\r\\n", "<br>");
```
Comment (sreynen) [inline markdown_posts/markdown_converter/Converter.java]: I don't think we actually want this line. This is converting line breaks to HTML, but the source is already HTML, so it's inserting new line breaks where they don't belong.

**Comment by sreynen** [comment]
Okay! I got it to run. I left a note inline on a specific line, but a more general note here: the output isn't yet formatted as markdown frontmatter. That seems worth getting it working before trying to process all the posts, which I expect will run much slower and introduce some new complications.

**Comment by Pyrrh** [comment]
Yep, that's what I'm trying to test out -- how well the conversion is working. I'm using a heavy helping of inline HTML rather than a full conversion to markdown where it may not be necessary. I had some commits and files on the other branch that I needed on this branch and merging everything together seemed like the fastest way to achieve that (in the future I may take a different route).

I'm on the same page with getting the conversion to work for a single post before expanding it but I just need everything all together to proceed.

File: markdown_posts/markdown_converter/Converter.java (hunk 1 above)
Comment (Pyrrh) [reply markdown_posts/markdown_converter/Converter.java]: I didn't notice any extra line breaks -- the \r\n single breaks are being replaced with a single <br>. Did you have a different experience when you ran the code? 

The reason I'm switching to <br> from the return + newline is due to documentation I read around use of HTML newline within markdown files. If the use of <br> does not work then I am planning to change to using markdown line breaks instead (two spaces at the end of a line) in place of the \r\n.

Example of use of <br> in markdown is here: https://stackoverflow.com/questions/13730982/force-line-break-br-in-header-h1-in-markdown

**Pull request merged**